src/popup/pages/NetworkForm.vue

```diff
@@ -1,243 +1,308 @@
 <template>
-  <Form v-slot="{ errors }">
+  <div
+    class="network-form"
+    data-cy="network-form"
+  >
+    <InfoBox
+      v-if="isNetworkPrefilled"
+      :type="INFO_BOX_TYPES.warning"
+      :text="$t('pages.network.thirdPartyDetails')"
+    />
+    <p
+      v-else
+      class="text-description"
+    >
+      {{ $t('pages.network.formLabel') }}
+    </p>
+
+    <Field
+      v-slot="{ field, errorMessage }"
+      key="name"
+      name="name"
+      :rules="{
+        required: true,
+        network_name: true,
+        network_exists: customNetworks,
+        max_len: NETWORK_NAME_MAX_LENGTH,
+      }"
+    >
+      <InputField
+        v-bind="field"
+        v-model="newNetworkName"
+        data-cy="network-name"
+        :placeholder="$t('pages.network.networkNamePlaceholder')"
+        :label="$t('pages.network.networkNameLabel')"
+        :message="errorMessage"
+        :text-limit="NETWORK_NAME_MAX_LENGTH"
+      />
+    </Field>
+
     <div
-      class="network-form"
-      data-cy="network-form"
+      v-for="{ inputs, name, protocol } in formStructure"
+      :key="protocol"
+      :data-cy="`group-${protocol}`"
     >
-      <InfoBox
-        v-if="isNetworkPrefilled"
-        :type="INFO_BOX_TYPES.warning"
-        :text="$t('pages.network.thirdPartyDetails')"
+      <hr>
+      <h3
+        class="text-heading-3"
+        v-text="name"
       />
-      <p
-        v-else
-        class="text-description"
-      >
-        {{ $t('pages.network.formLabel') }}
-      </p>
       <Field
-        v-for="input in formConfig"
+        v-for="input in inputs"
         v-slot="{ field, errorMessage }"
-        :key="input.key"
-        :name="input.key"
-        :rules="validatorRules(input.key)"
-        :validate-on-mount="fieldsToCheckOnMount!.includes(input.key)"
+        :key="protocol + input.key"
```

Comment:
why sometimes you use template strings and here simple concatenation - shouldn't we stick to one solution?

Reply:
`key` is only for the internal Vue logic - it doesn't need to be beautiful.
The `data-cy` is used in the tests so I'm concatenating it to add `-` between the chunks to make it more readable.
Does this makes sense?